tests/test_dataset_generator.py

```diff
@@ -0,0 +1,126 @@
+import unittest
+from unittest import mock
+
+import pandas as pd
+from numpy.random import PCG64, Generator
+
+from synthetic_data.dataset_generator import generate_dataset_by_class
+
+
+class TestDatasetGenerator(unittest.TestCase):
+    def setUp(self):
+        self.rng = Generator(PCG64(12345))
+        self.dataset_length = 10
+
+    def test_generate_dataset_with_invalid_generator(self):
+        columns_to_gen = [{"generator": "non existent generator"}]
+        with self.assertRaisesRegex(
+            ValueError, "generator: non existent generator is not a valid generator."
+        ):
+            generate_dataset_by_class(
+                self.rng,
+                columns_to_generate=columns_to_gen,
+                dataset_length=self.dataset_length,
+                path=None,
+            )
+
+    @mock.patch("synthetic_data.dataset_generator.logging.warning")
+    def test_generate_dataset_with_none_columns(self, mock_warning):
+        empty_dataframe = pd.DataFrame()
+        df = generate_dataset_by_class(self.rng, None, self.dataset_length, None)
+        mock_warning.assert_called_once_with(
+            "columns_to_generate is empty, empty dataframe will be returned."
+        )
+        self.assertEqual(empty_dataframe.empty, df.empty)
+
+    def test_generate_custom_dataset(self):
+        columns_to_gen = [
+            {"generator": "integer", "name": "int", "min_value": 4, "max_value": 88},
+            {
+                "generator": "datetime",
+                "name": "dat",
+                "start_date": pd.Timestamp(2001, 12, 22),
+                "end_date": pd.Timestamp(2022, 12, 22),
+            },
+            {
+                "generator": "text",
+                "name": "txt",
+                "chars": ["0", "1"],
+                "str_len_min": 300,
+                "str_len_max": 301,
+            },
+            {
+                "generator": "string",
+                "name": "str",
+                "chars": ["0", "1"],
+                "str_len_min": 2,
+                "str_len_max": 5,
+            },
+            {
+                "generator": "categorical",
+                "name": "cat",
+                "categories": ["X", "Y", "Z"],
+                "probabilities": [0.1, 0.5, 0.4],
+            },
+            {
+                "generator": "float",
+                "name": "flo",
+                "min_value": 3,
+                "max_value": 10,
+                "sig_figs": 3,
+            },
+        ]
```

Comment:
could probably move this to `setUp` and then use `self.columns_to_gen` in `test_path_to_csv`